ArcGISRuntimeSDKQt_CppSamples/UtilityNetwork/CreateLoadReport/CreateLoadReport.cpp

```diff
@@ -252,34 +252,28 @@ void CreateLoadReport::setUtilityTraceOrconditionWithCodedValue(CodedValue coded
   m_traceParameters->traceConfiguration()->traversability()->setBarriers(utilityTraceOrCondition);
 }
 
-void CreateLoadReport::createTraceCompletedConnection()
+void CreateLoadReport::onTraceCompleted_(const QString& codedValueName)
 {
-  connect(m_utilityNetwork, &UtilityNetwork::traceCompleted, this, [this](const QUuid& taskId)
-  {
-    const QString codedValueName = m_tasks.take(taskId);
-
-    UtilityTraceResultListModel* results = m_utilityNetwork->traceResult();
+  UtilityTraceResultListModel* results = m_utilityNetwork->traceResult();
 
-    for (UtilityTraceResult* result : *results)
-    {
-      // Get the total customers from the UtilityElementTraceResult
-      if (UtilityElementTraceResult* elementResult = dynamic_cast<UtilityElementTraceResult*>(result))
-        m_phaseCust[codedValueName] = elementResult->elements(this).size();
-
-      // Get the total load from the UtilityFunctionTraceResult
-      else if (UtilityFunctionTraceResult* functionResult = dynamic_cast<UtilityFunctionTraceResult*>(result))
-        m_phaseLoad[codedValueName] = qAsConst(functionResult)->functionOutputs().first()->result().toInt();
-    }
+  for (UtilityTraceResult* result : *results)
+  {
+    // Get the total customers from the UtilityElementTraceResult
+    if (UtilityElementTraceResult* elementResult = dynamic_cast<UtilityElementTraceResult*>(result))
+      m_phaseCust[codedValueName] = elementResult->elements(this).size();
 
-    emit loadReportUpdated();
+    // Get the total load from the UtilityFunctionTraceResult
+    else if (UtilityFunctionTraceResult* functionResult = dynamic_cast<UtilityFunctionTraceResult*>(result))
+      m_phaseLoad[codedValueName] = qAsConst(functionResult)->functionOutputs().first()->result().toInt();
+  }
 
-    // If the tasks queue is empty, all trace tasks have completed
-    if (m_tasks.isEmpty())
-    {
-      m_sampleStatus = CreateLoadReport::SampleReady;
-      emit sampleStatusChanged();
-    }
-  });
+  emit loadReportUpdated();
+  // If the trace request count is zero, all trace tasks have completed
+  if (!--m_traceRequestCount)
```

Comment:
```suggestion
  if ((--m_traceRequestCount) == 0)
```

Might be a bit more readable. Let me know.